src/main/java/duke/command/TaskList.java

```diff
@@ -0,0 +1,242 @@
+package duke.command;
+
+import java.io.IOException;
+
+import duke.exception.InvalidIndexException;
+import duke.exception.MissingContentException;
+import duke.task.Task;
+
+/**
+ * Handles data and formats into tasklist
+ */
+public class TaskList {
+    private String[] arr;
+
+    public TaskList() {
+        arr = new String[100];
+    }
+
+    /**
+     * Initialize a new TaskList object
+     * @param arr
+     */
+    public TaskList(String[] arr) {
+        this.arr = arr;
+    }
+
+    /**
+     * Returns the tasklist in array
+     * @return the tasklist in array
+     * @throws IOException if array does not exist
+     */
+    public String[] readTaskList() throws IOException {
+        try {
+            try {
+                return this.arr;
+            } catch (IndexOutOfBoundsException e) {
+                throw new MissingContentException();
+            }
+        } catch (MissingContentException e) {
+            System.out.println(e.getMessage());
+        }
+        return this.arr;
```

Comment:
I'm not entirely sure whether using nested try catch statements makes sense. An alternative would be using just one try catch statement, but in the catch statement you check for different types of errors and handle differently from there.